packages/Toast/src/Toast.js

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import CheckIcon from "@paprika/icon/lib/Check";
+import CautionIcon from "@paprika/icon/lib/Caution";
+import ExclamationCircleIcon from "@paprika/icon/lib/ExclamationCircle";
+import InfoCircleIcon from "@paprika/icon/lib/InfoCircle";
+import LockIcon from "@paprika/icon/lib/Lock";
+
+import Kinds from "./ToastKinds";
+
+import toastStyles, { contentStyles, IconStyled, CloseButtonStyled } from "./Toast.styles";
+
+const propTypes = {
+  /**
+   * Will make this an "assertive" alert that will interrupt immediately.
+   * Default behaviour is to be "polite" and wait until the user is idle.
+   */
+  ariaAlert: PropTypes.bool,
+
+  /** Will automatically close after 1500ms (or longer if provided by autoCloseDelay) */
+  canAutoClose: PropTypes.bool,
+
+  /** Duration (in ms) before Toast will automaticall close (if canAutoClose is true) */
+  autoCloseDelay: PropTypes.number,
+
+  /** Content of the Toast */
+  children: PropTypes.node,
+
+  /** If the component should have a 'close' button */
+  hasCloseButton: PropTypes.bool,
+
+  /** How "controlled" toast is shown / hidden. */
+  isOpen: PropTypes.bool,
+
+  /** Callback that is executed after clicking the 'close' button */
+  onClose: PropTypes.func,
+
+  /** If the Toast is fixed to the top of the viewport */
+  isFixed: PropTypes.bool,
+
+  /** Determines the styling of the Toast */
+  kind: PropTypes.oneOf([Kinds.SUCCESS, Kinds.WARNING, Kinds.ERROR, Kinds.INFO, Kinds.LOCKED, Kinds.VISUALLY_HIDDEN]),
+
+  /** The z-index of the Toast */
+  zIndex: PropTypes.number,
+};
+
+const defaultProps = {
+  ariaAlert: false,
+  canAutoClose: false,
+  autoCloseDelay: 1500,
+  children: null,
+  hasCloseButton: true,
+  isOpen: undefined,
+  onClose: () => {},
+  isFixed: false,
+  kind: Kinds.INFO,
+  zIndex: 1,
+};
+
+const minimumCloseTimeout = 1500;
+
+const icons = {
+  [Kinds.SUCCESS]: CheckIcon,
+  [Kinds.WARNING]: CautionIcon,
+  [Kinds.ERROR]: ExclamationCircleIcon,
+  [Kinds.INFO]: InfoCircleIcon,
+  [Kinds.LOCKED]: LockIcon,
+};
+
+function Toast(props) {
+  const {
+    ariaAlert,
+    canAutoClose,
+    autoCloseDelay,
+    children,
+    hasCloseButton,
+    isOpen,
+    onClose,
+    isFixed,
+    kind,
+    zIndex,
+    ...moreProps
+  } = props;
+
+  const ariaLive = ariaAlert ? "assertive" : "polite";
+  const [isToastOpen, setIsToastOpen] = React.useState(isOpen === undefined ? true : isOpen);
+  const timerRef = React.useRef(null);
+
+  function handleClose() {
+    clearTimeout(timerRef.current);
+    if (isOpen === undefined) setIsToastOpen(false);
+    onClose();
+  }
+
+  function startTimer() {
+    const newTimer = setTimeout(() => {
+      handleClose();
+    }, Math.max(autoCloseDelay, minimumCloseTimeout));
+
+    timerRef.current = newTimer;
+  }
+
+  const memoizedStartTimer = React.useCallback(startTimer, []);
```

Comment:
https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
> usually you’ll want to declare functions needed by an effect inside of it

Let's do that by inlining `startTimer` and moving `handleClose` into `useCallback`:
```js
const memoizedStartTimer = React.useCallback(() => {
  function handleClose() {
    clearTimeout(timerRef.current);
    if (isOpen === undefined) setIsToastOpen(false);
    onClose();
  }
  
  const newTimer = setTimeout(handleClose, Math.max(autoCloseDelay, minimumCloseTimeout));
  timerRef.current = newTimer;
}, []);
```
Now we can see that we've missed `autoCloseDelay`, `isOpen` and `onClose` as dependencies of `useCallback`